client/src/components/App/Content/InvestigationForm/TabManagement/PersonalInfoTab/PersonalInfoTab.tsx

```diff
@@ -47,210 +48,162 @@ const TRANSPORTATION_COMPANY_NAME_LABEL = 'שם החברה*';
+            [PersonalInfoDataContextFields.EDUCATION_CLASS_NUMBER, (value) => parseInt(value)]
+        ]);
+        return (specialConvertors.has(key)) ? 
+                    (specialConvertors.get(key) as DataParser)(value)
+                :
+                    value;
+    }
 
     const convertToDBData = (): PersonalInfoDbData => {
         const data = methods.getValues();
-        return {
-            phoneNumber: data.phoneNumber || null,
-            additionalPhoneNumber: data.additionalPhoneNumber || null,
-            contactPhoneNumber: data.contactPhoneNumber || null,
-            contactInfo: data.contactInfo || null,
-            insuranceCompany: data.insuranceCompany || null,
-            address: {
-                city: data.city || null,
-                street: data.street || null,
-                floor: data.floor || null,
-                houseNum: data.houseNum || null
-            },
-            relevantOccupation: data.relevantOccupation || null,
-            educationOccupationCity: data.educationOccupationCity || null,
-            institutionName: data.institutionName || null,
-            otherOccupationExtraInfo: data.otherOccupationExtraInfo || null,
-            role: data.role || null,
-            educationGrade: data.educationGrade || null,
-            educationClassNumber: data.educationClassNumber ? +data.educationClassNumber : null,
-        }
-    }
+        const dataKeys = Object.keys(data);
 
-    useEffect(() => {
-        const address: FlattenedDBAddress = {
-            city: data.city || null,
-            street: data.street || null,
-            floor: data.floor || null,
-            houseNum: data.houseNum || null,
-            streetName
-        }
-
-        setAddress(address);
-    }, [data.city, data.street, data.floor, data.houseNum]);
+        const parsedData = Object.values(data)
+                                .map(value => value || null)
+                                .reduce((obj, value, index) => {
+                                    const key = dataKeys[index] as keyof PersonalInfoTabState;
+                                    return Object.assign(obj, {
+                                        [key]: parseDataValue(key, value)
+                                    })
+                                }, {})
 
-    useEffect(() => {
-        if (data.role) {
-            setRoleObj(
-                investigatedPatientRoles.find((investigatedPatientRole: investigatedPatientRole) =>
-                    investigatedPatientRole.id === data.role) ?? defaultRole
-            );
+        parsedData.address = {
+            city: parsedData.city,
+            street: parsedData.street,
+            floor: parsedData.floor,
+            houseNum: parsedData.houseNum
         }
-    }, [investigatedPatientRoles, data.role]);
 
+        delete parsedData.city;
+        delete parsedData.street;
+        delete parsedData.floor;
+        delete parsedData.houseNum;
```

Comment:
Isn't it just simpler to create a new object without those fields? or changed the naming in the form to address.X?

Reply:
see [field arrays](https://react-hook-form.com/advanced-usage#FieldArrays) to use someting like address[city] or something like that